core/shared/src/main/scala/zio/stm/ZSTM.scala

```diff
@@ -157,6 +157,16 @@ final class ZSTM[-R, +E, +A] private[stm] (
   def fallback[A1 >: A](a: => A1)(implicit ev: CanFail[E]): ZSTM[R, Nothing, A1] =
     fold(_ => a, identity)
 
+  /**
+   * The moral equivalent of `if (p) exp`
+   */
+  def when[R1 <: R, E1 >: E](b: Boolean): ZSTM[R1, E1, Unit] = ZSTM.when(b)(self)
```

Comment:
We don't need the additional type parameters here since they aren't used anywhere.

Reply:
I wondered, but we also have them in https://github.com/zio/zio/blob/master/core/shared/src/main/scala/zio/ZIO.scala#L1559 and other places... should I also remove them there?

Reply:
removed